DM-18340: Update ATDomeTrajectory to use the target event from ATMCS

r-owen opened the pull request.

r-owen committed:
Use ATMCS instead of PointingComponent

Since it is a better source of the data and we aren't going to
build a PointingComponent anyway.

r-owen committed:
Modernize the code to set data in topics

instead of using DataType

r-owen committed:
git hide .coverage

patrickingraham commented on tests/test_dome_trajectory.py:
```diff
@@ -22,24 +22,21 @@
     def __init__(self, initial_state):
-        self.dome_index = 1  # match config/default.yaml
-        self.pointing_index = 2  # for Aux Tel
+        self.dome_index = 1  # 1=fixed, 2=portable
```
I think it would be useful to say why this index is needed for future developers (and me). It's not clear to me what you mean by portable, or when this would be used.

r-owen replied:
Agreed. I'll just say it `# match ts_ATDome`.

ATDome is indexed (which is fine; I wish all CSCs were) but it has two index enums: "fixed" = 1 and "portable" = 2. I have no idea what those mean. ts_ATDome uses 1.

patrickingraham commented on tests/test_dome_trajectory.py:
```diff
@@ -93,42 +89,35 @@ def test_standard_state_transitions(self):
-            start_data = harness.remote.cmd_start.DataType()
-            start_data.settingsToApply = "default.yaml"
-            await harness.remote.cmd_start.start(start_data, timeout=5)
+            harness.remote.cmd_start.set(settingsToApply="default.yaml")
```
The "test_standard_state_transitions" test is clear generalized for large numbers of CSCs. Is this something that's been made available to all Python developers? Is there a list of others? Seems like code everyone should recycle...

r-owen replied:
I agree it should be made standard. I think it is out of scope for this ticket but it would be a useful addition to ts_salobj.

patrickingraham commented on python/lsst/ts/ATDomeTrajectory/dome_trajectory.py:
```diff
@@ -73,20 +73,18 @@ def __init__(self, initial_state=salobj.base_csc.State.STANDBY, initial_simulati
-        self.pointing_remote = salobj.Remote(SALPY_PointingComponent, index=pointing_index,
-                                             include=["currentTargetStatus"])
+        self.atmcs_remote = salobj.Remote(SALPY_ATMCS, include=["target"])
         dome_index = 1  # I'm not sure, but I know it's fixed
```
I mentioned being confused by the dome_index in a comment in test_dome_trajectory.py where you mention portable versus fixed. I suppose I'm now more confused...

r-owen replied:
I agree. This was my original comment I found when I started this ticket it is not clear. I have changed them all as per above.

patrickingraham commented on python/lsst/ts/ATDomeTrajectory/dome_trajectory.py:
```diff
@@ -133,14 +131,13 @@ def begin_start(self, id_data):
 
-    async def update_current_target_status(self, current_target):
-        """Callback for currentTargetStatus from PointingComponent.
+    async def update_target(self, target):
```
line 121 mentions "This will be moved to BaseCsc at some point."
Would this be considered a support task? I'm asking because I'm supposed to review the completion of the development epic and I don't want this to get dropped as scope.

r-owen replied:
The issue here is that we have not yet decided how to support configuration of CSCs. We were converging on a solution when Tim A retired. Until we have a standard I don't want to put support in `ts_salobj`. I had to pick something for this CSC. It is technical debt that we will have to modify this code to match the standard.

patrickingraham commented on tests/test_dome_trajectory.py:
```diff
@@ -59,18 +56,17 @@ def test_main(self):
-                start_data.settingsToApply = "default.yaml"
-                await remote.cmd_start.start(start_data, timeout=5)
+                remote.cmd_start.set(settingsToApply="default.yaml")
+                await remote.cmd_start.start(timeout=5)
```
would the timeout values here ever change? I see you put 5's in multiple spots and was just thinking how much it'd be a pain if one day we decide state transitions must happen in less than X seconds etc.

r-owen replied:
Good suggestion. I've changed this to setting `STD_TIMEOUT ` at the top of the two tests that use timeouts. I also found 5 seconds was more than we needed so I changed to 2 seconds.

patrickingraham reviewed (commented):
Looks fine overall, but several comments/questions below. Many are not specific to changes in this PR.
Also, line 73 of algoritmms.py needs updating.

r-owen committed:
Fix comments for ATDome index

r-owen committed:
Use a constant for standard timeouts in tests

Also decrease standard timeout in test_dome_trajectory from 5 to 2 seconds, because 5 isn't needed.
